osu.Game/Skinning/Components/BeatmapAttributeText.cs

```diff
@@ -123,6 +145,132 @@ private void updateLabel()
+            valueDictionary[BeatmapAttribute.Creator] = beatmap.Value.BeatmapInfo.Metadata.Author.Username;
+            valueDictionary[BeatmapAttribute.Source] = beatmap.Value.BeatmapInfo.Metadata.Source;
+            valueDictionary[BeatmapAttribute.RankedStatus] = beatmap.Value.BeatmapInfo.Status.GetLocalisableDescription();
+        }
+
+        private void updateBeatmapDifficultyAttributes()
+        {
+            // Skip if not relevant attribute
+            if (!isBeatmapDifficultyAttribute(Attribute.Value))
+                return;
+
+            double rate = ModUtils.CalculateRateWithMods(mods.Value);
+
+            BeatmapDifficulty difficulty = new BeatmapDifficulty(beatmapInfo.Difficulty);
+
+            foreach (var mod in mods.Value.OfType<IApplicableToDifficulty>())
+                mod.ApplyToDifficulty(difficulty);
+
+            difficulty = ruleset.Value.CreateInstance().GetRateAdjustedDisplayDifficulty(difficulty, rate);
+
+            valueDictionary[BeatmapAttribute.CircleSize] = difficulty.CircleSize.ToLocalisableString(@"F2");
+            valueDictionary[BeatmapAttribute.HPDrain] = difficulty.DrainRate.ToLocalisableString(@"F2");
+            valueDictionary[BeatmapAttribute.Accuracy] = difficulty.OverallDifficulty.ToLocalisableString(@"F2");
+            valueDictionary[BeatmapAttribute.ApproachRate] = difficulty.ApproachRate.ToLocalisableString(@"F2");
+        }
+
+        private void updateTimingData()
+        {
+            // Skip if not relevant attribute
+            if (!isTimingData(Attribute.Value))
+                return;
+
+            double rate = ModUtils.CalculateRateWithMods(mods.Value);
+
+            valueDictionary[BeatmapAttribute.Length] = TimeSpan.FromMilliseconds(beatmapInfo.Length / rate).ToFormattedDuration();
+            valueDictionary[BeatmapAttribute.BPM] = FormatUtils.RoundBPM(beatmapInfo.BPM, rate).ToLocalisableString(@"F0");
+        }
+
+        private static bool isMetadata(BeatmapAttribute attribute) => (attribute >= BeatmapAttribute.Title && attribute <= BeatmapAttribute.Creator)
+                                                                      || attribute == BeatmapAttribute.RankedStatus || attribute == BeatmapAttribute.Source;
+
+        private static bool isBeatmapDifficultyAttribute(BeatmapAttribute attribute) => (attribute <= BeatmapAttribute.ApproachRate);
+
+        private static bool isTimingData(BeatmapAttribute attribute) => attribute == BeatmapAttribute.Length || attribute == BeatmapAttribute.BPM;
+
+        private static bool isDiffcalcInfo(BeatmapAttribute attribute) => attribute == BeatmapAttribute.StarRating || attribute == BeatmapAttribute.MaxPerformance;
+
+        #region diffcalc stuff
+
+        [Resolved]
+        private BeatmapDifficultyCache difficultyCache { get; set; } = null!;
+
+        private Bindable<StarDifficulty?>? bindableDifficulty;
```

Comment:
null null? no thanks

Reply:
`Bindable<StarDifficulty?>` is the return type of `difficultyCache.GetBindableDifficulty`
And this Bindable is not always set (it will be created only when Attribute is `StarRating` or `MaxPerformance`)
Because of this Bindable is nullable
How it can be made in a different way? I don't quite understand